app/controllers/api/v2/override_values_controller.rb

```diff
@@ -104,6 +105,11 @@ def rename_use_puppet_default
         params[:override_value][:omit] = params[:override_value].delete(:use_puppet_default)
         Foreman::Deprecation.api_deprecation_warning('"use_puppet_default" was renamed to "omit"')
       end
+
+      def validate_value
+        return unless params[:override_value] && params[:override_value].key?(:value)
+        render_error(:custom_error, :locals => {:message => _('value must be a string')}, :status => :bad_request) unless params[:override_value][:value].is_a?(String)
```

Comment:
This error can be misleading, it says `value must be a string`  when it can be boolean, integer, etc.. but contained in a string. I'm not sure how to word it, maybe `value must be contained in a string. e.g: if you want to send a boolean true, send 'true' as a string`.